File: app/src/main/java/com/sakuna63/tumbin/application/util/ViewStubProxyUtils.java
package com.sakuna63.tumbin.application.util;

import android.databinding.ViewStubProxy;
import android.support.annotation.NonNull;
import android.view.View;

public final class ViewStubProxyUtils {
    private ViewStubProxyUtils() {
    }

    public static <T> T inflate(@NonNull ViewStubProxy stubProxy) {
        if (!stubProxy.isInflated()) {
            stubProxy.getViewStub().inflate();
        }
        //noinspection unchecked
        return (T) stubProxy.getBinding();
    }


    public static void goneRootIfInflated(@NonNull ViewStubProxy stubProxy) {
        if (stubProxy.isInflated()) {
            stubProxy.getRoot().setVisibility(View.GONE);
        }
    }
}

File: app/src/main/java/com/sakuna63/tumbin/application/util/ViewStubProxyUtils.kt
package com.sakuna63.tumbin.application.util

import android.databinding.ViewStubProxy
import android.view.View

object ViewStubProxyUtils {

    fun <T> inflate(stubProxy: ViewStubProxy): T {
        if (!stubProxy.isInflated) {
            stubProxy.viewStub.inflate()
        }
        //noinspection unchecked
        return stubProxy.binding as T
    }


    fun goneRootIfInflated(stubProxy: ViewStubProxy) {
        if (stubProxy.isInflated) {
            stubProxy.root.visibility = View.GONE
        }
    }
}
